### ISSUE TITLE
Blasr run issue

### ISSUE THREAD

**Issue by madhurat** [PacificBiosciences/blasr#319]
Hello,
I am running Pbalign for PacBio assembly correction where BLASR v0.2.0 is being used in background by pbalign. While running blasr, I am getting an error saying "**BlasrService returned a non-zero exit status**". Command line arguments are correct but still it is showing the error.Can anyone please help me to solve this error???

**Comment by armintoepfer** [comment]
The current version is 5.3, are you sure that you are using version 0.2.0?

**Comment by ylipacbio** [comment]
We recommend to use the latest blasr and pbalign to work on both sequel and RSII data. For RSII data in HDF5 files, please first convert HDF5 files to bam files using bax2bam. 

However, if you have to work on HDF5 files for some reason, please install legacy blasr and legacy pbalign. 

```
git clone git@github.com:PacificBiosciences/pitchfork.git
cd pitchfork
make init
make legacy_pbalign PREFIX=/opt/mybuild             # if you want to build your own blasr
bash --init-file /opt/mybuild/setup-env.sh # either to use the build in the sub-shell
source /opt/mybuild/setup-env.sh           # or     to use the build in current shell 
```

**Comment by madhurat** [comment]
Thank you so much for your help. Blasr run issue has been solved but apparently samFilter is not running properly. Here is an error log:
**2016-12-23 16:04:31,997 [INFO] pbalign version: 0.2.0.141024
2016-12-23 16:04:37,259 [INFO] BlasrService: Align reads to references using /media/9759e2b0-edca-4aaa-a141-110bc7765ca3/Program/_SMART/smrtanalysis/current/analysis/bin/blasr.
2016-12-26 02:58:12,925 [INFO] FilterService: Filter alignments using /media/9759e2b0-edca-4aaa-a141-110bc7765ca3/Program/_SMART/smrtanalysis/current/analysis/bin/samFilter.
2016-12-26 02:58:15,764 [ERROR] FilterService returned a non-zero exit status. Segmentation fault (core dumped)**

Both Blasr and Samfilter executables are in the same working directory as Pbalign, but still samfilter is giving an error of segmentation fault (Enough space is available in the working directory). Please help us to resolve this problem. Really not getting what is to be done?

**Comment by madhurat** [comment]
Hello,
Thank you so much for your help. Blasr run issue has been solved but apparently samFilter is not running properly. Here is an error log:
2016-12-23 16:04:31,997 [INFO] pbalign version: 0.2.0.141024
2016-12-23 16:04:37,259 [INFO] BlasrService: Align reads to references using /media/9759e2b0-edca-4aaa-a141-110bc7765ca3/Program/_SMART/smrtanalysis/current/analysis/bin/blasr.
2016-12-26 02:58:12,925 [INFO] FilterService: Filter alignments using /media/9759e2b0-edca-4aaa-a141-110bc7765ca3/Program/_SMART/smrtanalysis/current/analysis/bin/samFilter.
2016-12-26 02:58:15,764 [ERROR] FilterService returned a non-zero exit status. Segmentation fault (core dumped)Both Blasr and Samfilter executables are in the same working directory as Pbalign, but still samfilter is giving an error of segmentation fault (Enough space is available in the working directory). Please help us to resolve this problem. Really not getting what is to be done?Regards,Madhura
 Madhura Tathode
Genome Artist       

Bionivid Technology Private Limited
3rd Floor, 4C-209| 4th Cross| Near New Horizon College|
Kasturi Nagar|Bangalore-560043 |INDIA
tel (+91) 9535619191| mobile (+91) 9916375730
website  |  map  |  email
   
Confidentiality Note: This email may contain confidential and/or private information. If you received this email in error please delete and notify sender.


 

    On Saturday, 24 December 2016 4:53 AM, Yuan Li <notifications@github.com> wrote:
 

 We recommend to use the latest blasr and pbalign to work on both sequel and RSII data. For RSII data in HDF5 files, please first convert HDF5 files to bam files using bax2bam.However, if you have to work on HDF5 files for some reason, please install legacy blasr and legacy pbalign.git clone git@github.com:PacificBiosciences/pitchfork.git
cd pitchfork
make init
make legacy_pbalign PREFIX=/opt/mybuild             # if you want to build your own blasr
bash --init-file /opt/mybuild/setup-env.sh # either to use the build in the sub-shell
source /opt/mybuild/setup-env.sh           # or     to use the build in current shell

**Comment by mjhsieh** [comment]
Are you using the smrtanalysis 2.3.0 or building legacy pbalign/blasr with pitchfork? And what did you mean by assembly correction? Was it "polishing" step of a pacbio assembly protocol or a pacbio reseqeuncing prototcol?

**Comment by madhurat** [comment]
Hi,
Yes, I am using pbalign/blasr with pitchfork and assembly step means polishing step of a pacbio assembly protocol I am running using Quiver.Average genome size is 2.6 GB and draft genome assembly is already done using Canu.Please help me to solve this error.It's really not getting solved.Thanks in advance!!

 Madhura Tathode
Genome Artist       

Bionivid Technology Private Limited
3rd Floor, 4C-209| 4th Cross| Near New Horizon College|
Kasturi Nagar|Bangalore-560043 |INDIA
tel (+91) 9535619191| mobile (+91) 9916375730
website  |  map  |  email
   
Confidentiality Note: This email may contain confidential and/or private information. If you received this email in error please delete and notify sender.


 

    On Friday, 30 December 2016 4:39 AM, Mengjuei Hsieh <notifications@github.com> wrote:
 

 Are you using the smrtanalysis 2.3.0 or building legacy pbalign/blasr with pitchfork? And what did you mean by assembly correction? Was it "polishing" step of a pacbio assembly protocol or a pacbio reseqeuncing prototcol?—

**Comment by mjhsieh** [comment]
From what I see from your log you are mixing smrtanalysis 2.3.0 with binaries from pitchfork build. We don't support this combination and we can't help you on this combination. Can you just use smrtanalysis 2.3.0 instead?

**Comment by mjhsieh** [comment]
There are many possible causes for samFilter to exit with segfault. You could modify your pbalign command (I presume that you are directly running pbalign command in a unix shell.) to add `--debug --verbose` option and see if you can isolate the exact samFilter command that failed.

**Comment by ylipacbio** [comment]
Read through this thread and found that samFilter segfault is caused by the fact that blasr v5.3  sam output files have been significantly changed which are not backward-compatible and cannot work with samFilter. 

As described in blasr wiki: 
https://github.com/PacificBiosciences/blasr/wiki/Legacy-blasr-for-RSII,-latest-blasr-for-SEQUEL-and-RSII

>PacBio RSII sequencing data used to be saved in BAX/BAS.H5 files. Although users can still use SMRTAnalysis v2.3- for analyzing RSII sequencing data, we recommend to first convert BAX/BAS.H5 files to BAM files and then analyze BAM files using SMRTAnalysis v3.0+, and the latest blasr and pbalign.

>As a result, pbalign and blasr has been significantly changed in SMRTAnalysis v3.0 (and later version), and are NOT BACKWARD-COMPATIBLE with blasr and pbalign in SMRTAnalysis v2.3 (and previous version). We call pbalign and blasr in SMRTAnalysis v2.3- legacy pbalign and legacy blasr.

So please convert your HDF files into bam and use the latest blasr/pbalign combo, or if you want to still use HDF files, stick with legacy blasr/pbalign (as included by smrtanalysis 2.3).

**Comment by ylipacbio** [comment]
Documented this issue in wiki: https://github.com/PacificBiosciences/blasr/wiki/BLASR-Frequently-Asked-Questions#why-samfilter-crashed-on-sam-file-created-by-blasr
